pkg/controller/compute/claim.go

```diff
@@ -103,8 +151,15 @@ func ConfigureAKSCluster(_ context.Context, cm resource.Claim, cs resource.NonPo
 	if spec.NodeCount == nil {
 		spec.NodeCount = to.IntPtr(v1alpha2.DefaultNodeCount)
 	}
-	spec.WriteServicePrincipalSecretTo = corev1.LocalObjectReference{Name: fmt.Sprintf("principal-%s", cm.GetUID())}
-	spec.WriteConnectionSecretToReference = corev1.LocalObjectReference{Name: string(cm.GetUID())}
+
+	spec.WriteServicePrincipalSecretTo = runtimev1alpha1.SecretReference{
```

Comment:
Do we feel good as a universal best practice to write *all* secrets produced by managed resources to the same namespace? I don't see any problems with it, I just wanted to bring up in case you could think of a situation where we would want different behavior.

Reply:
I'm not sure. I think this is the only managed resource that creates more than one secret, and it's not super obvious to me why it writes the service principal to a distinct secret from the connection secret. I suppose probably because it has to read it back? I could see the case that we should have a distinct `writeServicePrincipalSecretsToNamespace` class option for this case if we need to keep doing this, but I'd rather tackle that in future.